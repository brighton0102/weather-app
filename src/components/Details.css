.weather-card-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 20px 0;
}

.second-card {
  background-color: rgba(202, 69, 111, 1);
  width: 100%;
  padding: 20px 0;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  border-radius: 5px;
}

.card img {
  max-width: 100%;
}

.details-container {
  text-align: center;
}

.second-title {
  font-family: 'Raleway', sans-serif;
  color: white;
  font-size: 20px;
  padding-top: 10px;
}

.loading-spinner {
  border: 4px solid rgba(0, 0, 0, 0.3);
  border-top: 4px solid black;
  border-radius: 50%;
  width: 24px;
  height: 24px;
  animation: spin 1s linear infinite;
  position: relative;
  left: 48%;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

@media screen and (min-width: 768px) {
  .second-card {
    width: 320px;
  }
}
