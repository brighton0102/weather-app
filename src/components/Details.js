import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import fetchWeatherByCityName from '../services/api';
import './Details.css';

function Details() {
  const { city } = useParams();
  const [weatherData, setWeatherData] = useState(null);

  const loadWeatherData = async (cityName) => {
    try {
      const data = await fetchWeatherByCityName(cityName);
      setWeatherData(data);
    } catch (error) {
      console.error(error);
    }
  };

  useEffect(() => {
    // Load weather data for the selected city when the component mounts
    loadWeatherData(city);
  }, [city]);

  return (
    <div className="details-container">
      <p className="second-title">
        {city}
        <br />
        live weather report
      </p>
      {weatherData ? (
        <div className="weather-card-content">
          <div className="second-card">
            <img src={weatherData.current.condition.icon} alt="Weather Condition" />
            <p>
              Condition:
              {weatherData.current.condition.text}
            </p>
          </div>
          <div className="second-card">
            <p>
              Temperature:
              {weatherData.current.temp_c}
              °C (
              {weatherData.current.temp_f}
              °F)
            </p>
          </div>
          <div className="second-card">
            <p>
              Humidity:
              {weatherData.current.humidity}
              %
            </p>
          </div>
          <div className="second-card">
            <p>
              Wind Speed:
              {' '}
              {weatherData.current.wind_kph}
              {' '}
              km/h
            </p>
          </div>
          <div className="second-card">
            <p>
              Wind Degree:
              {weatherData.current.wind_degree}
              °
            </p>
          </div>
          <div className="second-card">
            <p>
              Wind Direction:
              {weatherData.current.wind_dir}
            </p>
          </div>
          <div className="second-card">
            <p>
              Cloud Cover:
              {weatherData.current.cloud}
              %
            </p>
          </div>
          <div className="second-card">
            <p>
              Feels Like:
              {' '}
              {weatherData.current.feelslike_c}
              °C (
              {weatherData.current.feelslike_f}
              °F)
            </p>
          </div>
          <div className="second-card">
            <p>
              Visibility:
              {' '}
              {weatherData.current.vis_km}
              {' '}
              km (
              {weatherData.current.vis_miles}
              {' '}
              miles)
            </p>
          </div>
          <div className="second-card">
            <p>
              UV Index:
              {weatherData.current.uv}
            </p>
          </div>
          <div className="second-card">
            <p>
              Wind Gust:
              {' '}
              {weatherData.current.gust_kph}
              {' '}
              km/h (
              {weatherData.current.gust_mph}
              {' '}
              mph)
            </p>
          </div>
        </div>
      ) : (
        <div className="loading-spinner" />
      )}
    </div>

  );
}

export default Details;
