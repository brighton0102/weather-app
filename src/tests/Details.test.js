import React from 'react';
import { render } from '@testing-library/react';
import Details from '../components/Details';

test('Details component renders correctly', () => {
  const { container } = render(<Details />);
  expect(container).toMatchSnapshot();
});
