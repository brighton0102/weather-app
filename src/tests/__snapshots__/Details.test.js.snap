// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders correctly 1`] = `
<div>
  <div
    class="details-container"
  >
    <p
      class="second-title"
    >
      <br />
      live weather report
    </p>
    <div
      class="loading-spinner"
    />
  </div>
</div>
`;
